src/redisjson.rs

```diff
@@ -218,17 +218,25 @@ impl RedisJSON {
         &self,
         path: &str,
         scalar: &str,
-        start: usize,
-        end: usize,
+        start: i64,
+        end: i64,
     ) -> Result<i64, Error> {
         if let Value::Array(arr) = self.get_doc(path)? {
+            if arr.len() <= 0 || end < -1 {
```

Comment:
Why `<= 0`? The length can't be negative; should be `== 0`.
And then it would be better to use `arr.is_empty()`.

Reply:
done